### PR TITLE
Time-lock example

### PR DESCRIPTION
**Pull request by BlindDev**
Hey, everyone.

This is our first attempt in Scrypto and RUST. 

In this example, we added an ability to lock XRD with a certain duration (will be added to the current epoch).

### CODE REVIEW

**Comment by BlindDev** [comment]
Made some changes for the blueprint. Right now the lock will use badges to handle the locked amount

File: defi/time-lock/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            
+            // Put fees in collected XRD.
+            self.collected_fees.put(fee_tokens);
+
+            // Mint TL badge with locked amount and end epoch as metadata
+            let tl_resource_def = ResourceBuilder::new()
+                .metadata("name", "Time lock badge")
+                .metadata("amount", available.to_string())
+                .metadata("ends", end_time.to_string())
+                .new_badge_mutable(self.tl_minter_vault.resource_def());
```
Comment (iamyulong) [inline defi/time-lock/src/lib.rs:74]: This would be a perfect candidate for NFT with Scrypto v0.2.0.

**Comment by iamyulong** [comment]
I'm merging it now, but feel free to make another PR for enhancements.

**Comment by BlindDev** [comment]
Hey @iamyulong, thank you so much. Yes, I was hoping to use some new Scrypto features for all projects I uploaded

**Pull request merged**